docs/user/adding-new-metrics.md

```diff
@@ -50,7 +50,32 @@ The details of the metric parameters are described in [metric parameters](metric
 
 The `metrics.yaml` file is used to generate `Kotlin` code that becomes the public API to access your application's metrics.
 
-## Metric naming
+## Recommendations for defining new metrics
+
+“There are only two hard things in Computer Science: cache invalidation and naming things” -- attributed to Phil Karlton.
+
+### Lifetimes
+
+The `lifetime` parameter of a metric defines when it will be reset. There are three options available:
+
+- `ping` (default): The metric is reset each time it is sent in the ping.
+  This is the most common case, and should be used for metrics that are highly dynamic, such as things computed in response to the user's interaction with the application.
+- `user`: The metric is part of the user's profile.
+  This should be used for things that change only when the user's profile is created.
+  It is rare to use this lifetime outside of some metrics that are built in to Glean, such as `client_id`.
+- `application`: The metric is related to an application run, and is reset only when the application restarts.
```

Comment:
IMO I would order them by their ... lifetime.
ping < application < user

Reply:
+1 on this

Reply:
Good idea.